packages/cozy-stack-client/src/OAuthClient.js

```diff
@@ -0,0 +1,358 @@
+      })
+    )
+    
+    this.oauthOptions = this.camelCaseOAuthData({
+      ...this.oauthOptions,
+      client_id: data.client_id,
+      client_name: data.client_name,
+      client_secret: data.client_secret,
+      registration_access_token: data.registration_access_token,
+      software_id: data.software_id
+    })
+
+    return this.oauthOptions
+  }
+
+  /**
+   * Unregisters the currenly configured client with the OAuth server. 
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @returns {promise}
+   */
+  async unregister() {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+
+    return this.fetch(
+      'DELETE',
+      `/auth/register/${this.oauthOptions.clientID}`,
+      null,
+      {
+        credentials: this.registrationAccessTokenToAuthHeader()
+      }
+    )
+  }
+
+  /**
+   * Fetches the complete set of client informations from the server after it has been registered.
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @returns {promise}
+   */
+  async fetchInformations() {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+
+    return this.fetch(
+      'GET',
+      `/auth/register/${this.oauthOptions.clientID}`,
+      null,
+      {
+        credentials: this.registrationAccessTokenToAuthHeader()
+      }
+    )
+  }
+
+  /**
+   * Updates the client own information. This method will update both the local information and the remote information on the OAuth server.
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @param   {object} informations Set of informations to update. Note that some fields such as `clientID` can't be updated.
+   * @param   {boolean} resetSecret = false Optionnal, whether to reset the client secret or not
+   * @returns {promise} A promise that resolves to a complete, updated list of client informations
+   */
+  async updateInformations(informations, resetSecret = false) {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+
+    const mandatoryFields = {
+      clientID: this.oauthOptions.clientID,
+      clientName: this.oauthOptions.clientName,
+      redirectURI: this.oauthOptions.redirectURI,
+      softwareID: this.oauthOptions.softwareID
+    }
+    const data = this.snakeCaseOAuthData({
+      ...mandatoryFields,
+      ...informations
+    })
+
+    if (resetSecret) data['client_secret'] = this.oauthOptions.clientSecret
+
+    const result = await this.fetch(
+      'PUT',
+      `/auth/register/${this.oauthOptions.clientID}`,
+      data,
+      {
+        credentials: this.registrationAccessTokenToAuthHeader()
+      }
+    )
+
+    this.oauthOptions = this.camelCaseOAuthData(result)
+
+    return this.oauthOptions
+  }
+
+  /**
+   * Generates a random state code to be used during the OAuth process
+   * @returns {string}
+   */
+  generateStateCode() {
```

Comment:
Replacing @ptbrowne comment from [here](https://github.com/cozy/cozy-client/pull/18#discussion_r179415626)

> It is a bit confusing to see state here. generateNone ? > https://fr.wikipedia.org/wiki/Nonce_(cryptographie) 

I found the name very confusing too, but it's called `state` on the stack as well](https://github.com/cozy/cozy-stack/blob/master/docs/auth.md#get-authauthorize) so I think it's better to keep the same name. And I've called it `stateCode` everywhere, instead of simply `state`.

Reply:
If it's confusing, let's change it also in the stack ?

Reply:
Yeah that could be a thing. @nono or @jinroh do you see any reason not to do it (renaming `state` to `nonce` in the oauth process) ?

Reply:
I see a good reason to not do that: [the RFC about OAuth 2.0](https://tools.ietf.org/html/rfc6749#section-4.1.1) says that this parameter is named `state`